upgrade_versions.py

```diff
@@ -239,6 +239,8 @@ def push_new_versions(versions_repo, release_date, versions_repo_push_url,
     refspec = "HEAD:refs/heads/{}".format(versions_repo_push_branch)
     push_info = remote.push(refspec=refspec)[0]
     LOG.debug("Push result: {}".format(push_info.summary))
+    if git.PushInfo.ERROR & push_info.flags:
+        raise repository.PushError(push_info)
 
```

Comment:
This could be sqaushed with @35b0a414f53e5d99ceebfc14bfb96836a3dbcbfa but I'll not make a hard time about this either, we'll get into the meaning of atomicity and such all over again.